### ISSUE TITLE
Support loose (invalid) JavaScript

### ISSUE THREAD

**Issue by remcohaszing** [mdx-js/mdx-analyzer#267]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

While typing, code is often invalid. Various IntelliSense features typically work on invalid code. TypeScript supports this, but acorn does not. Currently invalid syntax is caught, but IntelliSense doesn’t work. We need to find a way to deal with invalid syntax.

### Solution

Needs investigation

### Alternatives

:shrug:

**Comment by ChristianMurphy** [comment]
Maybe https://github.com/acornjs/acorn/tree/master/acorn-loose/ would be of interest?

**Comment by wooorm** [comment]
the micromark utils do a low an acorn instance to be passed, so that might also indeed allow the loose version? Worth to check, but might not be strong enough

**Comment by remcohaszing** [comment]
I think `acorn-loose` is a good candidate, although currently it doesn’t work as a drop-in replacement, because it doesn’t implement `parseExpressionAt`.

```js
import {Parser} from 'acorn'
import {LooseParser} from 'acorn-loose'
import jsx from 'acorn-jsx'
import remarkMdx from 'remark-mdx'
import remarkParse from 'remark-parse'
import {unified} from 'unified'

// const parser = Parser.extend(jsx())
const parser = LooseParser.extend(jsx())

const processor = unified().use(remarkParse).use(remarkMdx, {acorn: parser})

const mdx = `
import { Button } from './button.js'

Click the button

<Button>Click me!</Button>
`

console.dir(processor.parse(mdx))
```

---

Also worth considering is [`@typescript-eslint/typescript-estree`](https://typescript-eslint.io/architecture/typescript-estree/), or possibly even just TypeScript, as we’re not really interested in the estree.
